### ISSUE TITLE
Help wanted: Using with pagination 

### ISSUE THREAD

**Issue by TonyVeigel** [erikras/redux-form#222]
Scenario: I have a page that a user submits a form and returns the first 20 images. I then want to use pagination to call the server for each additional 20 images. Problem is if they happen to change information on the form and paginate, they would be doing a whole new search because the form data changed. Do I need to set up another state for this scenario? Or could I do something like initializing the form on handle submit so that the initial value would be my second state:

 ``` 
handleSubmit(data){
    dispatch(initialize('imageSearch', data));
    dispatch(searchImages(data, page));
  }

handlePaginate(){
//get initial values and submit in case they changed the form
}
```
then when they paginate I would use initial values but when they submit I would use the data given to me. 

or am I way off base and shouldn't be using redux form for this.

Thanks for any insight.

**Comment by erikras** [comment]
@TonyVeigel Did you solve this? I've been working older issues today and hadn't gotten to address yours. If you solve it, could you share how?

**Comment by TonyVeigel** [comment]
I just used the way I described above, haven't tested it yet but should work. If it is the right way to do it I don't know, kind of new to all this. 

Every time they submit I update the initial state of the form. That way if they change the form data and then paginate (server side if we already don't have the data in the store), I can use the initial values instead of the current values. If they submit the form, I just use the data provided. I didn't see a easy way to get the key value pairs of the initial data so I just extracted it myself. 

```
  handleSubmit(data){
    const { dispatch, selectedImagePage} = this.props;
    dispatch(initialize('imageSearch', data));
    dispatch(searchImages(data, selectedImagePage)))
  }

  handlePagination(){
    const { dispatch, selectedImagePage} = this.props;
    dispatch(searchImagesIfNeeded(this.getInitialValuesOfForm(), selectedImagePage));
  }
```
